### PR TITLE
Np 44782 delete import candidate event consumer

### PR DESCRIPTION
**Pull request by truhacevkir**
Have implemented two event handlers:

DeleteImportCandidateEventConsumer: triggered by event sent by DeleteImportCandidateEventEmitter

DeleteImportCandidateEventHandler: triggered by event sent by ImportCandidateDataEntryUpdateHandler when newImage is null

The deletion flow for import candidates is now following:

DeleteImportCandidateEventEmitter -> 
DeleteImportCandidateEventConsumer ->
ImportCandidateDataEntryUpdateHandler -> 
DeleteImportCandidateEventHandler ->
DeleteImportCandidateFromIndex (SearchApi)

### CODE REVIEW

File: publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/delete/DeleteImportCandidateEventConsumer.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+            .orElseThrow();
+        return null;
+    }
+
+    private static ExpandedImportCandidate toExpandedImportCandidate(ImportCandidateSearchApiResponse response)
+        throws BadGatewayException {
+        if (containsMultipleHits(response)) {
+            throw new BadGatewayException(COULD_NOT_FETCH_UNIQUE_IMPORT_CANDIDATE_MESSAGE);
+        }
+        return response.getHits().get(UNIQUE_HIT_FROM_SEARCH_API);
```
Comment (brinxmat) [inline publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/delete/DeleteImportCandidateEventConsumer.java:63]: NPE waiting to happen…

File: publication-event-handlers/src/test/java/no/unit/nva/publication/events/handlers/delete/DeleteImportCandidateEventConsumerTest.java [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+        when(uriRetriever.getRawContent(any(), any())).thenReturn(toResponse(ExpandedImportCandidate.fromImportCandidate(importCandidate),
+                                                                             SINGLE_HIT));
+        handler.handleRequest(event, output, CONTEXT);
+        assertThrows(NotFoundException.class,
+                     () -> resourceService.getImportCandidateByIdentifier(importCandidate.getIdentifier()));
+        assertThat(appender.getMessages(), containsString(IMPORT_CANDIDATE_HAS_BEEN_DELETED_MESSAGE));
+    }
+
+    @Test
+    void shouldThrowBadGatewayExceptionWhenMultipleHitsInResponseFetchingUniqueImportCandidate()
```
Comment (brinxmat) [inline publication-event-handlers/src/test/java/no/unit/nva/publication/events/handlers/delete/DeleteImportCandidateEventConsumerTest.java:94]: Can you test for deleting a non-existing thing?

File: template.yaml [hunk 3]
```diff
@@ -1134,6 +1175,38 @@ Resources:
...
 
+  DeleteImportCandidateEventHandler:
+    DependsOn: EventsLambdaPermissions
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: publication-event-handlers
+      Handler: no.unit.nva.publication.events.handlers.delete.DeleteImportCandidateEventHandler::handleRequest
+      Runtime: java11
+      Role: !GetAtt LambdaRole.Arn
+      ReservedConcurrentExecutions: 64
```
Comment (brinxmat) [inline template.yaml:1186]: Excessive?

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: template.yaml (hunk 3 above)
Comment (truhacevkir) [reply template.yaml:1186]: Should I change reserved concurrency for all lambdas in deletion flow, or only this one? It is the same amount as ExpansionHandler uses and this one does similar action, i.e. its sends event to searchApi.

File: s3-import-commons/src/main/java/no/unit/nva/publication/s3imports/DeleteImportCandidatesEventEmitter.java [hunk 4]
```diff
@@ -24,14 +21,15 @@ public class DeleteImportCandidatesEventEmitter implements RequestHandler<S3Even
...
+    public static final String NOT_EMITTED_EVENTS = "Not emitted events {}:";
     private static final int SINGLE_EXPECTED_FILE = 0;
     private static final Logger logger = LoggerFactory.getLogger(DeleteImportCandidatesEventEmitter.class);
     private final S3Client s3Client;
     private final EventBridgeClient eventBridgeClient;
 
     @JacocoGenerated
     public DeleteImportCandidatesEventEmitter() {
-        this(S3Driver.defaultS3Client().build(), defaultEventBridgeClient());
+        this(S3Driver.defaultS3Client().build(), ApplicationConstants.defaultEventBridgeClient());
```
Comment (anetteOlli) [inline s3-import-commons/src/main/java/no/unit/nva/publication/s3imports/DeleteImportCandidatesEventEmitter.java:32]: Something is telling me that the "ApplicationConstants" class does a bunch of environment reads. Consider swapping it for your own method if possible.

File: template.yaml (hunk 3 above)
Comment (anetteOlli) [reply template.yaml:1186]: I think 1 should be enough. 
If we have to resort to 64 concurrent executions for deleting a bunch of scopus-identifiers, it's time to consider a sqs queue or simply dumping the database.

**Review by anetteOlli — APPROVED** [review]

**Pull request merged**